File: WorldFeed/Server/src/WorldFeed.Common/Public/Parameters/Optionals/GetCursorUsersOptionalParameters.cs
namespace WorldFeed.Common.Public.Parameters.Optionals
{
    using WorldFeed.Common.Public.Parameters;

    /// <inheritdoc />
    public interface IGetCursorUsersOptionalParameters : ICursorQueryParameters, IGetUsersOptionalParameters
    {
    }
    
    /// <inheritdoc />
    public class GetCursorUsersOptionalParameters : CursorQueryParameters, IGetUsersOptionalParameters
    {
        public GetCursorUsersOptionalParameters()
        {
        }

        public GetCursorUsersOptionalParameters(IGetCursorUsersOptionalParameters parameters)
            : base(parameters)
        {
            IncludeEntities = parameters?.IncludeEntities;
            SkipStatus = parameters?.SkipStatus;
        }        
        
        /// <inheritdoc />
        public bool? IncludeEntities { get; set; }
        /// <inheritdoc />
        public bool? SkipStatus { get; set; }
    }
}

File: WorldFeed/Client/src/app/core/Public/Parameters/Optionals/GetCursorUsersOptionalParameters.ts
import {CursorQueryParameters, ICursorQueryParameters} from "../CursorQueryParameters";
import {IGetUsersOptionalParameters} from "./GetUsersOptionalParameters";

export interface IGetCursorUsersOptionalParameters extends ICursorQueryParameters, IGetUsersOptionalParameters {
}

export class GetCursorUsersOptionalParameters extends CursorQueryParameters implements IGetUsersOptionalParameters {
  constructor(parameters?: IGetCursorUsersOptionalParameters) {
    if (parameters) {
      super(parameters);

      this.IncludeEntities = parameters?.IncludeEntities;
      this.SkipStatus = parameters?.SkipStatus;
    } else {
      super();
    }
  }

  public IncludeEntities?: boolean;
  public SkipStatus?: boolean;
}

// public GetCursorUsersOptionalParameters()
// {
// }
//
// public GetCursorUsersOptionalParameters(IGetCursorUsersOptionalParameters parameters) : base(parameters)
// {
//   IncludeEntities = parameters?.IncludeEntities;
//   SkipStatus = parameters?.SkipStatus;
// }
